### PR TITLE
Added option to control devices by device name

### PR DESCRIPTION
**Pull request by kwazel**
Since I have no functional DNS server in my home network and I refuse to remember IP-addresses, I found the possibility to reach devices using their device name (as set using the smartphone App) convenient

### CODE REVIEW

File: pyHS100/cli.py [hunk 1]
```diff
@@ -80,6 +89,25 @@ def discover(ctx, timeout, discover_only):
...
+    click.echo("Trying to discover %s using %s attempts of %s seconds" %
+               (devicename, attempts, timeout))
+    for attempt in range(1, attempts):
+        print("Attempt %s of %s" % (attempt, attempts))
+        found_devs = Discover.discover(timeout=timeout).items()
+        for ip, dev in found_devs:
+            if dev.alias.lower() == devicename.lower():
+                host = dev.host
+                return host
+    return None
```
Comment (rytilahti) [inline pyHS100/cli.py]: Should this rather raise a corresponding exception instead of returning none?

File: pyHS100/cli.py [hunk 2]
```diff
@@ -80,6 +89,25 @@ def discover(ctx, timeout, discover_only):
...
+@click.option('--timeout', default=3, required=False)
+@click.option('--attempts', default=10, required=False)
+@click.option('--devicename', required=True)
+@click.pass_context
+def find_host_from_device_name(ctx, timeout, attempts, devicename):
+    """Discover devices in the network."""
+    host = None
+    click.echo("Trying to discover %s using %s attempts of %s seconds" %
+               (devicename, attempts, timeout))
+    for attempt in range(1, attempts):
```
Comment (rytilahti) [inline pyHS100/cli.py:102]: Is there really a need to do multiple attempts? When does that happen? 10 attempts with timeout of 3 seconds sounds a bit much, doesn't it?

File: pyHS100/cli.py [hunk 3]
```diff
@@ -80,6 +89,25 @@ def discover(ctx, timeout, discover_only):
     return found_devs
 
 
+@cli.command()
+@click.option('--timeout', default=3, required=False)
+@click.option('--attempts', default=10, required=False)
+@click.option('--devicename', required=True)
+@click.pass_context
+def find_host_from_device_name(ctx, timeout, attempts, devicename):
```
Comment (rytilahti) [inline pyHS100/cli.py:97]: Does this have to be a command? Is there some usecase where someone would call this from console directly?

**Review by rytilahti — COMMENTED** [review]
Although I would probably prefer a configuration file by myself (that would avoid doing the discovery in those cases where the IP address stays the same), I can see how this can be useful and would be fine merging this after the comments are handled.

File: pyHS100/cli.py (hunk 2 above)
Comment (kirichkov) [reply pyHS100/cli.py:102]: I agree with @rytilahti. I think a default timeout of 1 second and 3 attempts sounds acceptable to me.

File: pyHS100/cli.py (hunk 1 above)
Comment (kirichkov) [reply pyHS100/cli.py]: I see the logic of returning `None`. If a device can't be found via name, the cli will continue searching for a device using discovery. On the other hand unsuccessful discovery also doesn't raise an exception, so I think we should adhere here to `discover`'s behavior.

**Pull request closed without merging**